metaflow/plugins/aws/awslambda/lambda_decorator.py

```diff
@@ -0,0 +1,213 @@
+from metaflow.plugins.aws.awslambda.lambda_runner import LAMBDA_MAX_MEMORY_MB
+import sys
+from typing import TYPE_CHECKING, Optional, List, Dict
+from metaflow.decorators import StepDecorator, Decorator
+from metaflow.plugins.timeout_decorator import get_run_time_limit_for_task
+from metaflow import R
+from metaflow.datastore.datastore import TransformableObject
+from metaflow.exception import MetaflowException
+from metaflow.plugins.aws.sync_metadata import sync_metadata_to_s3
+from metaflow.plugins.aws.batch.batch_decorator import (
+    BatchDecorator,
+    DEFAULT_TIMEOUT_SECONDS,
+    ResourcesDecorator,
+)
+from metaflow.plugins.conda.conda_step_decorator import CondaStepDecorator
+
+if TYPE_CHECKING:
+    from metaflow.runtime import CLIArgs
+    from metaflow.graph import FlowGraph
+    from metaflow.package import MetaflowPackage
+    from metaflow.flowspec import FlowSpec
+    from metaflow.datastore import MetaflowDataStore
+    from metaflow.environment import MetaflowEnvironment
+    from metaflow.event_logger import EventLogger
+
+from metaflow.metaflow_config import (
+    LAMBDA_LOG_GROUP_NAME,
+    LAMBDA_ROLE_ARN,
+    LAMBDA_IMAGE_URI,
+)
+
+# Supported resource specs for Lambda
+LAMBDA_MAX_RESOURCES = {
+    "memory": LAMBDA_MAX_MEMORY_MB,
+    "cpu": 6,
+}
+DEFAULT_TIMEOUT_SECONDS = 600
+
+
+class LambdaException(MetaflowException):
+    headline = "AWS Lambda error"
+
+
+class LambdaDecorator(StepDecorator):
+    """
+    The decorator is called @awslambda to avoid confusion, since "lambda" is a reserved
+    word in Python.
+    """
+
+    name = "awslambda"
```

Comment:
thoughts on using `@lambda` instead of `@awslambda` to keep it consistent with `@batch`?

Reply:
That's what i had at first but `lambda` is a reserved word in Python